generate_cert.go

```diff
@@ -137,6 +139,11 @@ func main() {
 		template.KeyUsage |= x509.KeyUsageCertSign
 	}
 
+	if *isDC {
+		template.ExtraExtensions = append(
+			template.ExtraExtensions, *tls.CreateDelegationUsagePKIXExtension())
```

Comment:
Maybe instead of this, which is a new pattern, you add a field to x509.Certificate called PermitsDelegatedCredentials and rig up the machinery to create the extension for you.

Reply:
crypto/x509 is not part of tris, just crypto/tls

Reply:
Ok, then just keep it in mind in case you're asked to upstream

Reply:
Done. I'm adding a NOTE here about this at this line.